packages/xstate-vue/src/useSelector.ts

```diff
@@ -12,10 +12,22 @@ const noop = () => {
 export function useSelector<TActor extends ActorRef<any, any>, T>(
   actor: TActor | Ref<TActor>,
   selector: (snapshot: SnapshotFrom<TActor>) => T,
-  compare: (a: T, b: T) => boolean = defaultCompare
-) {
+  compare?: (a: T, b: T) => boolean
+): Ref<T>;
+export function useSelector<TActor extends ActorRef<any, any>, T>(
+  actor: (TActor | undefined) | Ref<TActor | undefined>,
+  selector: (snapshot: SnapshotFrom<TActor> | undefined) => T,
+  compare?: (a: T | undefined, b: T) => boolean
+): Ref<T | undefined>;
+export function useSelector<TActor extends ActorRef<any, any>, T>(
+  actor: (TActor | undefined) | Ref<TActor | undefined>,
+  selector: (snapshot: SnapshotFrom<TActor> | undefined) => T,
+  compare: (a: T | undefined, b: T) => boolean = defaultCompare
+): Ref<T | undefined> {
   const actorRefRef = isRef(actor) ? actor : shallowRef(actor);
-  const selected = shallowRef(selector(actorRefRef.value.getSnapshot()));
+  const selected = shallowRef(
+    actorRefRef.value ? selector(actorRefRef.value.getSnapshot()) : undefined
+  );
```

Comment:
This is not correct because we want `selector` to be called with `undefined` if the user doesn't pass `actor` in. Otherwise, you won't get `T` back.

This kinda gets patched immediately by the `watch` call that follows this since it uses `immediate: true`. I'm not sure if `watch` callbacks would be called server-side though - so it's better to fix this here to not rely on this implicit "fixup"

Reply:
I just pushed out a fix for this